bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationHandler.java

```diff
@@ -0,0 +1,438 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeliosVentilationHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class HeliosVentilationHandler extends BaseThingHandler implements SerialPortEventListener {
+    private static final int BUSMEMBER_MAINBOARD = 0x11;
+    private static final int BUSMEMBER_SLAVEBOARDS = 0x10;
+    private static final byte BUSMEMBER_CONTROLBOARDS = (byte) 0x20;
+    private static final int BUSMEMBER_REC_MASK = 0xF0; // interpreting frames delivered to BUSMEMBER_ME &
+                                                        // BUSMEMBER_REC_MASK
+    private static final int BUSMEMBER_ME = 0x2F; // used as sender when communicating with the helios system
+    private static final int POLL_OFFLINE_THRESHOLD = 3;
+
+    /** Logger Instance */
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private @NonNullByDefault({}) HeliosVentilationConfiguration config;
+
+    private SerialPortManager serialPortManager;
+
+    private @Nullable SerialPort serialPort;
+    private @Nullable InputStream inputStream;
+    private @Nullable OutputStream outputStream;
+
+    private @Nullable ScheduledFuture<?> pollingTask;
+    private int pollCounter;
+
+    /**
+     * store received data for read-modify-write operations on bitlevel
+     */
+    private HashMap<Byte, Byte> memory = new HashMap<Byte, Byte>();
+
+    public HeliosVentilationHandler(Thing thing, final SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(HeliosVentilationConfiguration.class);
+
+        logger.debug("Serial Port: {}, 9600 baud, PollPeriod: {}", config.serialPort, config.pollPeriod);
+
+        if (config.serialPort.length() < 1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Port must be set!");
+            return;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            if (this.config.pollPeriod > 0) {
+                startPolling();
+            }
+        }
+
+        scheduler.execute(this::connect);
+    }
+
+    private synchronized void connect() {
+        logger.debug("HeliosVentilation: connecting...");
+        // parse ports and if the port is found, initialize the reader
+        SerialPortIdentifier portId = serialPortManager.getIdentifier(config.serialPort);
+        if (portId == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Port " + config.serialPort + " is not known!");
+            serialPort = null;
+            IOUtils.closeQuietly(inputStream);
```

Comment:
We are trying to migrate away from using the apache util libraries. Can you implement this without using IOUtils?